Marlin/src/HAL/DUE/Servo.cpp

```diff
@@ -71,33 +71,33 @@ void Servo_Handler(const timer16_Sequence_t, Tc*, const uint8_t);
 #endif
 
 void Servo_Handler(const timer16_Sequence_t timer, Tc *tc, const uint8_t channel) {
-  // clear interrupt
-  tc->TC_CHANNEL[channel].TC_SR;
-  if (Channel[timer] < 0)
-    tc->TC_CHANNEL[channel].TC_CCR |= TC_CCR_SWTRG; // channel set to -1 indicated that refresh interval completed so reset the timer
-  else if (SERVO_INDEX(timer, Channel[timer]) < ServoCount && SERVO(timer, Channel[timer]).Pin.isActive)
-    extDigitalWrite(SERVO(timer, Channel[timer]).Pin.nbr, LOW); // pulse this channel low if activated
-
-  Channel[timer]++;    // increment to the next channel
-  if (SERVO_INDEX(timer, Channel[timer]) < ServoCount && Channel[timer] < SERVOS_PER_TIMER) {
-    tc->TC_CHANNEL[channel].TC_RA = tc->TC_CHANNEL[channel].TC_CV + SERVO(timer,Channel[timer]).ticks;
-    if (SERVO(timer,Channel[timer]).Pin.isActive)    // check if activated
-      extDigitalWrite(SERVO(timer, Channel[timer]).Pin.nbr, HIGH); // its an active channel so pulse it high
+  tc->TC_CHANNEL[channel].TC_SR;                                      // clear interrupt
+  int8_t cho = Channel[timer];                                        // Handle the prior Channel[timer] first
+  if (cho < 0)                                                        // Channel -1 indicates the refresh interval completed...
+    tc->TC_CHANNEL[channel].TC_CCR |= TC_CCR_SWTRG;                   // ...so reset the timer
+  else if (SERVO_INDEX(timer, cho) < ServoCount && SERVO(timer, cho).Pin.isActive)  // prior channel... handled and active?
+    extDigitalWrite(SERVO(timer, cho).Pin.nbr, LOW);                  // pulse the prior channel LOW
+
+  Channel[timer] = ++cho;                                             // go to the next channel (or 0)
+  if (SERVO_INDEX(timer, cho) < ServoCount && cho < SERVOS_PER_TIMER) {
+    tc->TC_CHANNEL[channel].TC_RA = tc->TC_CHANNEL[channel].TC_CV + SERVO(timer, cho).ticks;
+    if (SERVO(timer, cho).Pin.isActive)                               // activated?
+      extDigitalWrite(SERVO(timer, cho).Pin.nbr, HIGH);               // yes: pulse HIGH
   }
   else {
     // finished all channels so wait for the refresh period to expire before starting over
-    tc->TC_CHANNEL[channel].TC_RA =
-      tc->TC_CHANNEL[channel].TC_CV < usToTicks(REFRESH_INTERVAL) - 4
-        ? (unsigned int)usToTicks(REFRESH_INTERVAL) // allow a few ticks to ensure the next OCR1A not missed
-        : tc->TC_CHANNEL[channel].TC_CV + 4;        // at least REFRESH_INTERVAL has elapsed
-    Channel[timer] = -1; // this will get incremented at the end of the refresh period to start again at the first channel
+    const unsigned int cval = tc->TC_CHANNEL[channel].TC_CV + 4,      // at least REFRESH_INTERVAL has elapsed
```

Comment:
The `+4` here won't function as intended with the faster clock rate. Previously 4 ticks was about 128 microseconds, now it is only 8.

Maybe 8 is long enough, but it would probably be a good idea to replace the `4` with something that is actually time-based, so that it will adjust if prescaler changes.

The intent of this is just to make sure the trigger level can't be set in the past if there are pulses going all the way to the end of the 20ms period.

Reply:
What about this line in `initISR`? Should 2625 be changed to 42000 get 1ms delay with MCK/2?
```cpp
  // 84MHz, MCK/32, for 1.5ms: 3937
  TC_SetRA(tc, channel, 2625); // 1ms = 84000000UL / 32UL / 1000UL
```

Reply:
Yes, that should probably be switched to be calculated using the prescaler also, rather than just updating the constant.

That is just setting the interrupt to fire a safe time in the future before enabling it, since the first actual pulse is turned on inside the interrupt.